### ISSUE TITLE
Prefer "var" doesn't fire if you assign from a delegate field.

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#23711]
![image](https://user-images.githubusercontent.com/1103906/33819782-4f09b35e-dea0-11e7-9197-611c2861b052.png)

``` C#
    class Program
    {
        static void Main(string[] args)
        {
            GetHandler handler = Handler; // <-- Expected "IDE0007"

        }

        private static GetHandler Handler;

        delegate object GetHandler();
    }
```

**Comment by zaytsev-victor** [comment]
@davkean your example works as expected because ```handler``` and ```Handler``` have different types.
But I found that IDE0007 is missed in the following example

```csharp
class Program
{
    static void Main(string[] args)
    {
        GetHandler handler = Handler;
    }

    private static GetHandler Handler;

    delegate object GetHandler();
}
```

**Comment by davkean** [comment]
@zaytsev-victor Not sure I understand. Are you saying that IDE0007 is firing for you? Or are you suggesting that it shouldn't fire even though I've told the IDE that I always want "var"?

**Comment by davkean** [comment]
Ah, I understand. You are correct. My example was mixed up (I'd cut it down from another example, and it matches your case.

**Comment by jcouv** [comment]
Fix is merged for 15.6 preview 4
